### PR TITLE
Refactor logging to use new `VS_LOG_AND_FLUSH` and `VS_LOG` macros

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [X] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Please list any related issues
- Implements #455 

Purpose:
- What is this pull request trying to do? Refactor Vega Strike's logging, as we have previously discussed
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to? 0.9.x

### CODE REVIEW

File: engine/src/vs_math.h [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+ *
+ * This file is part of Vega Strike.
  *
- * This program is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * as published by the Free Software Foundation; either version 2
- * of the License, or (at your option) any later version.
+ * Vega Strike is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
```
Comment (stephengtuggy) [inline engine/src/vs_math.h:14]: Whoops. I'll change that back to `version 2`

File: engine/src/vs_globals.h [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+ * Copyright (C) 2020-2021 pyramid3d, Stephen G. Tuggy, and
+ * other Vega Strike contributors
+ *
+ * https://github.com/vegastrike/Vega-Strike-Engine-Source
+ *
+ * This file is part of Vega Strike.
+ *
+ * Vega Strike is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
```
Comment (stephengtuggy) [inline engine/src/vs_globals.h:14]: I will fix this as well. Etc.

**Comment by stephengtuggy** [comment]
Hold up. I just discovered that the log files are going to a different directory than expected. I will need to do a bit more work on that.

**Comment by Loki1950** [comment]
Just be glad you caught it ;) macro's always seem to twist away from the intended behaviour.

**Comment by stephengtuggy** [comment]
OK, I think I finally have it.

**Comment by stephengtuggy** [comment]
Sorry about the size of this PR, guys. I know it's a lot. Hopefully, most of the changes will be fairly straightforward to review.

File: engine/src/vsfilesystem.h [hunk 3]
```diff
@@ -30,20 +31,22 @@
...
-#include <boost/log/sinks/sync_frontend.hpp>
-#include <boost/log/sinks/text_file_backend.hpp>
-#include <boost/log/sinks/text_ostream_backend.hpp>
+// #include <boost/smart_ptr/shared_ptr.hpp>
+// #include <boost/format.hpp>
+// #include <boost/log/core.hpp>
+// #include <boost/log/trivial.hpp>
+// #include <boost/log/sinks/sync_frontend.hpp>
+// #include <boost/log/sinks/text_file_backend.hpp>
+// #include <boost/log/sinks/text_ostream_backend.hpp>
```
Comment (BenjamenMeyer) [inline engine/src/vsfilesystem.h:47]: let's clean these up

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
Generally looking good; needs some cleanup and one assignment is missing the r-value.

File: engine/src/aldrv/al_init.cpp [hunk 4]
```diff
@@ -159,8 +160,8 @@ static ALCcontext *context_id = NULL;
 #endif
 bool AUDInit()
 {
-    g_game.sound_enabled =
 #ifdef HAVE_AL
+    g_game.sound_enabled =
```
[the remark this answers is not in the document]
Comment (stephengtuggy) [reply engine/src/aldrv/al_init.cpp]: fixed. This value is assigned again further down, so this line was unnecessary, AFAICT. Just deleted it.

File: engine/src/main.cpp [hunk 5]
```diff
@@ -78,42 +78,22 @@
...
-#include <boost/filesystem.hpp>
+#include "vs_logging.h"
 #include "options.h"
 
-using std::cout;
-using std::cerr;
-using std::endl;
+// using std::cout;
+// using std::cerr;
+// using std::endl;
```
[the remark this answers is not in the document]
Comment (stephengtuggy) [reply engine/src/main.cpp]: Done

File: engine/src/vsfilesystem.h (hunk 3 above)
Comment (stephengtuggy) [reply engine/src/vsfilesystem.h]: Done

**Review by BenjamenMeyer — APPROVED** [review]

**Pull request merged**